src/NetherPortalScanner.cpp

```diff
@@ -0,0 +1,290 @@
+
+#include "Globals.h"  // NOTE: MSVC stupidness requires this to be the same across all modules
+
+#include "NetherPortalScanner.h"
+#include "Vector3.h"
+#include "Entities/Entity.h"
+#include "World.h"
+
+
+
+
+
+cNetherPortalScanner::cNetherPortalScanner(cEntity * a_MovingEntity, cWorld * a_DestinationWorld, Vector3d a_DestPosition, int a_MaxY) :
+	m_Entity(a_MovingEntity),
+	m_World(a_DestinationWorld),
+	m_FoundPortal(false),
+	m_BuildPlatform(true),
+	m_Dir(Direction::X),
+	m_PortalLoc(a_DestPosition.Floor()),
+	m_Position(a_DestPosition),
+	m_MaxY(a_MaxY)
+{
+	int MinX = FloorC((m_Position.x - SearchRadius) / cChunkDef::Width);
+	int MinZ = FloorC((m_Position.z - SearchRadius) / cChunkDef::Width);
+	int MaxX = CeilC((m_Position.x + SearchRadius) / cChunkDef::Width);
+	int MaxZ = CeilC((m_Position.z + SearchRadius) / cChunkDef::Width);
+	for (int x = MinX; x < MaxX; x++)
+	{
+		for (int z = MinZ; z < MaxZ; z++)
+		{
+			Add(x, z);
+		}
+	}
+	Enable(*a_DestinationWorld->GetChunkMap());
+}
+
+
+
+
+
+void cNetherPortalScanner::OnChunkAvailable(int a_ChunkX, int a_ChunkZ)
+{
+	cChunkDef::BlockTypes blocks;
+	m_World->GetChunkBlockTypes(a_ChunkX, a_ChunkZ, blocks);
+
+	// Iterate through all of the blocks in the chunk
+	for (unsigned int i = 0; i < cChunkDef::NumBlocks; i++)
```

Comment:
Because of the way the chunks are stored (XZY?), doing i+2 here would result in a pattern that skips every other row. Note that it's iterating from 0 to NumBlocks, not 0 to 16 each axis - this results in a speedup in not having to call CoordinateToIndex.